packages/webview_flutter/webview_flutter_android/android/src/test/java/io/flutter/plugins/webviewflutter/WebViewClientTest.java

```diff
@@ -66,51 +53,9 @@ public void tearDown() {
     instanceManager.stopFinalizationListener();
   }
 
-  @Test
-  public void onPageStarted() {
-    webViewClient.onPageStarted(mockWebView, "https://www.google.com", null);
-    verify(mockFlutterApi)
-        .onPageStarted(eq(webViewClient), eq(mockWebView), eq("https://www.google.com"), any());
-  }
-
-  @Test
-  public void onReceivedError() {
-    webViewClient.onReceivedError(mockWebView, 32, "description", "https://www.google.com");
-    verify(mockFlutterApi)
-        .onReceivedError(
-            eq(webViewClient),
-            eq(mockWebView),
-            eq(32L),
-            eq("description"),
-            eq("https://www.google.com"),
-            any());
-  }
-
-  @Test
-  public void urlLoading() {
-    webViewClient.shouldOverrideUrlLoading(mockWebView, "https://www.google.com");
-    verify(mockFlutterApi)
-        .urlLoading(eq(webViewClient), eq(mockWebView), eq("https://www.google.com"), any());
-  }
-
-  @Test
-  public void convertWebResourceRequestWithNullHeaders() {
-    final Uri mockUri = mock(Uri.class);
-    when(mockUri.toString()).thenReturn("");
-
-    final WebResourceRequest mockRequest = mock(WebResourceRequest.class);
-    when(mockRequest.getMethod()).thenReturn("method");
-    when(mockRequest.getUrl()).thenReturn(mockUri);
-    when(mockRequest.isForMainFrame()).thenReturn(true);
-    when(mockRequest.getRequestHeaders()).thenReturn(null);
-
-    final GeneratedAndroidWebView.WebResourceRequestData data =
-        WebViewClientFlutterApiImpl.createWebResourceRequestData(mockRequest);
-    assertEquals(data.getRequestHeaders(), new HashMap<String, String>());
-  }
-
   @Test
   public void setReturnValueForShouldOverrideUrlLoading() {
```

Comment:
Since it looks like this PR is creating separate tests files for `WebViewClient` and `WebViewClientCompat`, you can go ahead and copy this file to the new tests and then delete this file.

Reply:
I actually left this one out on purpose, as this test is not dependent on an implementation.
This way there's 3 files with clear distinction
* `WebViewClientTest` which tests things that are not dependent on the implementation
* `WebViewClientImplTest` and `WebViewClientCompatImplTest` which test the 2 implementations.

Reply:
Yes, but both implementations have `setReturnValueForShouldOverrideUrlLoading`, right. So I think both files and implementations should have there own test for it.

Reply:
I've had a look at the implementation (I only looked at the test before) and now see that the internal implementation is indeed dependant on the `instanceof` check.  So there would indeed be a point of testing both.

I also noticed that `WebViewClientCompatImplTest` still had the test, so `WebViewClientTest` was redundant anyway.

Adding the test to `WebViewClientImplTest` is not possible with the current setup as I can't mock the static call to `Build.VERSION.SDK_INT`.

I have removed the redundant `WebViewClientTest`. The 'missing' test for `WebViewClientImplTest` was already missing and not directly related to my PR.